src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -785,14 +690,13 @@ set_post_data_from_params(CURL *curl, struct sol_arena *arena,
 
     SOL_VECTOR_FOREACH_IDX (&params->params, iter, idx) {
         struct sol_str_slice value = SOL_STR_SLICE_EMPTY;
-
         if (iter->type == SOL_HTTP_PARAM_POST_FIELD) {
             SOL_WRN("SOL_HTTP_PARAM_POST_FIELD and SOL_HTTP_PARAM_POST_DATA found in parameters."
                 " Only one can be used at a time");
             return false;
         } else if (iter->type == SOL_HTTP_PARAM_HEADER) {
-            const char *key = iter->value.key_value.key;
-            type_set = type_set || !strcasecmp(key, "content-type");
+            struct sol_str_slice key = iter->value.key_value.key;
+            type_set = type_set || !strncasecmp(key.data, "content-type", key.len);
```

Comment:
maybe we should have a `sol_str_slice_casecmp()` (and `sol_str_slice_caseeq()` on top, to be used here)?

And you should also check for size if you're using `strncasecmp()`, otherwise a shorter key will match:

`strncasecmp("content", "content-type", strlen("content")) == 0`, something we don't want.